pallets/schemas/src/lib.rs

```diff
@@ -328,228 +473,369 @@ pub mod pallet {
+		#[pallet::weight(T::WeightInfo::propose_to_create_intent_group())]
+		pub fn propose_to_create_intent_group(
+			origin: OriginFor<T>,
+			intent_group_name: crate::types::SchemaNamePayload,
+			intent_ids: BoundedVec<IntentId, T::MaxIntentsPerIntentGroup>,
+		) -> DispatchResult {
+			let proposer = ensure_signed(origin)?;
+
+			let proposal: Box<T::Proposal> = Box::new(
+				(Call::<T>::create_intent_group_via_governance {
+					creator_key: proposer.clone(),
+					intent_group_name,
+					intent_ids,
+				})
+				.into(),
+			);
+			T::ProposalProvider::propose_with_simple_majority(proposer, proposal)?;
 			Ok(())
 		}
 
-		/// Adds a given schema to storage. The schema in question must be of length
-		/// between the min and max model size allowed for schemas (see pallet
-		/// constants above). If the pallet's maximum schema limit has been
-		/// fulfilled by the time this extrinsic is called, a SchemaCountOverflow error
-		/// will be thrown.
+		/// Update an IntentGroup (testnet)
 		///
 		/// # Events
-		/// * [`Event::SchemaCreated`]
-		/// * [`Event::SchemaNameCreated`]
+		/// * [`Event::IntentGroupUpdated`]
 		///
 		/// # Errors
-		/// * [`Error::LessThanMinSchemaModelBytes`] - The schema's length is less than the minimum schema length
-		/// * [`Error::ExceedsMaxSchemaModelBytes`] - The schema's length is greater than the maximum schema length
-		/// * [`Error::InvalidSchema`] - Schema is malformed in some way
-		/// * [`Error::SchemaCountOverflow`] - The schema count has exceeded its bounds
-		/// * [`Error::InvalidSetting`] - Invalid setting is provided
-		/// * [`Error::InvalidSchemaNameEncoding`] - The schema name has invalid encoding
-		/// * [`Error::InvalidSchemaNameCharacters`] - The schema name has invalid characters
-		/// * [`Error::InvalidSchemaNameStructure`] - The schema name has invalid structure
-		/// * [`Error::InvalidSchemaNameLength`] - The schema name has invalid length
-		/// * [`Error::InvalidSchemaNamespaceLength`] - The schema namespace has invalid length
-		/// * [`Error::InvalidSchemaDescriptorLength`] - The schema descriptor has invalid length
-		/// * [`Error::ExceedsMaxNumberOfVersions`] - The schema name reached max number of versions
-		///
-		#[pallet::call_index(7)]
-		#[pallet::weight(
-			match schema_name {
-				Some(_) => T::WeightInfo::create_schema_v3_with_name(model.len() as u32 + settings.len() as u32),
-				None => T::WeightInfo::create_schema_v3_without_name(model.len() as u32 + settings.len() as u32)
-			}
-		)]
-		pub fn create_schema_v3(
+		/// * [`Error::InvalidIntentGroupId`] - The specified [`IntentGroupId`] does not exist.
+		/// * [`Error::InvalidIntentId`] - At least one of the specified [`IntentId`]s does not exist.
+		/// * [`Error::TooManyIntentsInGroup`] - The update would result in too many Intents in the group.
+		#[pallet::call_index(16)]
+		#[pallet::weight(T::WeightInfo::update_intent_group(intent_ids.len() as u32))]
+		pub fn update_intent_group(
```

Comment:
These extrinsics don't so much update as replace the intent group ids with the new collection, and that isn't obvious from the design doc / readme.